### DISCUSSION TITLE
How to show OpenCV output from function continuously running in background into .CTkFrame

### DISCUSSION THREAD

**Discussion by siddheshgunjal** [TomSchimansky/CustomTkinter#1120] [Q&A]
I Have a Function which runs prediction on video stream from camera (I have captured frames from `video_threading_optimization.py` for better performance). This function runs for 10 seconds after specific intervals. It keeps running in the background and captures data based on intervals set by user.
Following is the function for predictions which opens OpenCV default window:

```python
def predict_from_camera(detector, sec):
    # print('Press q to exit')
    vs = WebcamVideoStream(src=0).start()
    fps = FPS().start()
    window_not_set = True

    date_ = []
    time_ = []
    names_ = []
    predictions= []

    start = time.time()
    end = sec

    while break_program == False:

        # grab the frame from the threaded video stream
        image = vs.read()
        [h, w] = image.shape[:2]
        image = cv2.flip(image, 1)

        boxes, scores, classes, num_detections, output = detector.run(image)

        text = "classes: {}".format(output)
        cv2.putText(image, text, org=(20, 20), fontFace=cv2.FONT_HERSHEY_DUPLEX,
                    fontScale=0.5, color=(255, 0, 0))

        vis_util.visualize_boxes_and_labels_on_image_array(
            image,
            np.squeeze(boxes),
            np.squeeze(classes).astype(np.int32),
            np.squeeze(scores),
            category_index,
            use_normalized_coordinates=True,
            line_thickness=1)

        if window_not_set is True:
            cv2.namedWindow("Emotion Prediction Prototype (%d, %d)" % (720, 480), cv2.WINDOW_NORMAL)
            window_not_set = False

        cv2.imshow("Emotion Prediction Prototype (%d, %d)" % (720, 480), image)
        k = cv2.waitKey(1) & 0xff

        if time.time()-start > end:
            break
        fps.update()

    # stop the timer
    fps.stop()

    # do a bit of cleanup
    cv2.destroyAllWindows()
    vs.stop()
```

Following is the code for UI I have setup (Inspired from CustomTkinter examples):

```python
# create capture status frame
self.capture_status_frame = customtkinter.CTkFrame(self, corner_radius=4, fg_color="transparent")
self.capture_status_frame.grid_columnconfigure(0, weight=1)

# capture status title
self.capture_status_title = customtkinter.CTkLabel(self.capture_status_frame, text="Capture Status", font=font_heading)
self.capture_status_title.grid(row=0, column=0, columnspan=2, padx=20, pady=10)

# capture field
self.capture_field = customtkinter.CTkFrame(self.capture_status_frame, corner_radius=10, height=320, width=480)
self.capture_field.grid(row=1, column=0, columnspan=2, padx=20, pady=20)

#capture field text
self.capture_field_text = customtkinter.CTkLabel(self.capture_field, text="Not Capturing", font=font_details)
self.capture_field_text.place(relx=0.5, rely=0.5, anchor="center")
```
This is what it looks like:
![screen1_edit](https://user-images.githubusercontent.com/37741186/215096074-b3f3f3be-0981-4dcb-9cfb-e9801dfc14e0.png)

I have tried some of the examples to show openCV video in tkinter but those won't work for me as I will have to keep running the predictions in background even when UI is closed. So I cannot run function inside CustomTkinter Class. 

How Can we capture the frame from some function continuously running in background and show its frame inside CustomTkinter.CTkFrame whenever UI is launched?

Even a small help in the right direction would be appreciated. Thanks!

**Comment by sumeshir26** [comment]
Try using diffrent threads

**Reply by siddheshgunjal** [reply]
How can I do that? because even if I run it on different threads will the CTk class will fetch the frame continuously whenever function runs which is outside of its class?

**Reply by sumeshir26** [reply]
> How can I do that? because even if I run it on different threads will the CTk class will fetch the frame continuously whenever function runs which is outside of its class?

Yes, it will

**Comment by ElectricCandlelight** [comment]
Just seen this are you still struggling with this? I just implemented displaying a webcam feed from OpenCV to tkinter today.

Out of curiosity why do you need the video stream running even if the UI is closed?
